babble/src/main/java/io/mosaicnetworks/babble/servicediscovery/ResolvedGroup.java

```diff
@@ -24,12 +24,102 @@
 
 package io.mosaicnetworks.babble.servicediscovery;
 
+import java.util.ArrayList;
 import java.util.List;
 
-public interface ResolvedGroup {
-    void addService(ResolvedService resolvedService);
-    boolean removeService(ResolvedService resolvedService);
-    List<ResolvedService> getResolvedServices();
-    String getGroupName();
-    String getGroupUid();
+/**
+ * This represents a group of resolved services with the same group UID. Services with a matching
+ * group UID and group name can be added to the group as they're discovered. Services can be removed
+ * from the group as and when they are lost.
+ */
+public class ResolvedGroup {
+
+    public enum Source {
+        MDNS,
+        WEBRTC
+    }
+
```

Comment:
Note to self, there are some interesting methods I added to ResolvedGroup in my other branch.